### DISCUSSION TITLE
How to prevent `\renewcommand{\bibsection}{}` from being included in generated tex?

### DISCUSSION THREAD

**Discussion by ramiromagno** [quarto-dev/quarto-cli#6230] [Q&A] [answered]
### Description

If I place the bibliography with the special div:

```md
::: {#refs}
:::
```

I get in .tex:

```tex
\renewcommand{\bibsection}{}
\bibliography{wileyNJD-AMA}
```

If I omit that div altogether, my .tex template makes sure the bibliography is added right before the end:

```
$preamble.tex()$

\begin{document}

$title.tex()$

\maketitle

$body$

$biblio.tex()$

\end{document}
```

And I get in .tex:

```tex
\bibliography{wileyNJD-AMA}
```

And that is desirable in my case because of Wiley's class I am working with...

So, how can I make sure that `::: {#refs}` adds only `\bibliography{wileyNJD-AMA}` while leaving out `\renewcommand{\bibsection}{}`?

**Comment by cscheid** [comment]
> So, how can I make sure that `::: {#refs}` adds only `\bibliography{wileyNJD-AMA}` while leaving out `\renewcommand{\bibsection}{}`?

If I understand the source of that LaTeX command correctly, that's currently not possible: we hardcode it [here](https://github.com/quarto-dev/quarto-cli/blob/main/src/format/pdf/format-pdf.ts#L766).

We currently add that entry if [we detect the use of the `natbib` citation package](https://github.com/quarto-dev/quarto-cli/blob/main/src/format/pdf/format-pdf.ts#L323).

I would like to add more fine-grained control over our postprocessors, but this is currently low in our priority and we have our plates pretty full.

The easiest workaround for you to at the moment is probably to `quarto render --to latex` and then write a little `sed` script to drop that from the file.

**Comment by ramiromagno** [comment]
Thanks a lot for your answer!

I don't think the sed post-processing solution is viable because this is an extension for a journal template. AFAIK, you can't automate that kind of post-processing in your extension, can you?

**Reply by cscheid** [reply]
> I don't think the sed post-processing solution is viable because this is an extension for a journal template

It certainly won't be as convenient, but you can make your journal custom template produce a .tex file instead of a .pdf. Then you'll need to manually run your post-processing script yourself before producing the .pdf (again, manually).

**Reply by ramiromagno** [reply]
Maybe I don't yet fully understand how extensions work. But if someone else were to install and use my extension, that means he or she would have still to run that manually...?

**Reply by cscheid** [reply]
Currently, yes.

**Comment by ramiromagno** [comment, marked as answer]
As a workaround use a shortcode:

```
\bibliography{ {{< meta bibliography >}} }
```
